packages/fluentui/react/src/components/Icon/SvgIcon.tsx

```diff
@@ -0,0 +1,197 @@
+  setStart();
+
+  const {
+    alt,
+    'aria-label': ariaLabel,
+    bordered,
+    circular,
+    className,
+    disabled,
+    design,
+    children,
+    outline,
+    rotate,
+    size,
+    styles,
+    variables,
+    xSpacing
+  } = props;
+
+  const { classes } = useStyles(SvgIcon.displayName, {
+    className: SvgIcon.className,
+    mapPropsToStyles: () => ({
+      bordered,
+      circular,
+      disabled,
+      outline,
+      rotate,
+      size,
+      xSpacing
+    }),
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl
+  });
+
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(SvgIcon.handledProps, props);
+
+  const element = (
+    <ElementType
+      role="img"
+      aria-hidden={alt || ariaLabel ? undefined : 'true'}
+      aria-label={ariaLabel}
+      className={classes.root}
+      {...unhandledProps}
+    >
+      {callable(children)({ classes, rtl: context.rtl, props })}
+    </ElementType>
+  );
+  setEnd();
+
+  return element;
+};
+
+SvgIcon.className = 'ui-icon';
+SvgIcon.displayName = 'SvgIcon';
+SvgIcon.defaultProps = {
+  as: 'span',
+  size: 'medium',
+  rotate: 0
+};
+
+SvgIcon.propTypes = {
+  ...commonPropTypes.createCommon({
+    content: false
+  }),
+  children: PropTypes.func,
+  bordered: PropTypes.bool,
+  circular: PropTypes.bool,
+  disabled: PropTypes.bool,
+  outline: PropTypes.bool,
+  rotate: PropTypes.number,
+  size: customPropTypes.size,
+  xSpacing: PropTypes.oneOf(['none', 'before', 'after', 'both'])
+};
+SvgIcon.handledProps = Object.keys(SvgIcon.propTypes) as any;
+
+SvgIcon.create = ({ svg, displayName, handledProps }: SvgIconCreateFnParams) => {
+  // TODO: add props types for the component
+  const Component: any = (props: Extendable<SvgIconProps>) => {
+    // TODO: fix typings
+    const context: ProviderContextPrepared = React.useContext(ThemeContext);
+
+    const {
+      alt,
+      'aria-label': ariaLabel,
+      bordered,
+      circular,
+      className,
+      disabled,
+      design,
+      outline,
+      rotate = 0,
+      size = 'medium',
+      styles,
+      variables,
+      xSpacing
+    } = props;
+
+    const { classes } = useStyles(SvgIcon.displayName, {
+      className: SvgIcon.className,
+      mapPropsToStyles: () => ({
+        bordered,
+        circular,
+        disabled,
+        outline,
+        rotate,
+        size,
+        xSpacing
+      }),
+      mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+      rtl: context.rtl
+    });
+
+    const unhandledProps = getUnhandledProps(handledProps, props);
+
+    return (
+      <span
+        role="img"
+        aria-hidden={alt || ariaLabel ? undefined : 'true'}
+        aria-label={ariaLabel}
+        className={classes.root}
+        {...unhandledProps}
+      >
+        {callable(svg)({ classes, rtl: context.rtl, props })}
```

Comment:
```suggestion
        {svg({ classes, rtl: context.rtl, props })}
```

_nit_
I would say that it should be always a function